tests/src/python/test_qgsgeometry.py

```diff
@@ -6292,6 +6292,13 @@ def coerce_to_wkt(wkt, type, defaultZ=None, defaultM=None):
                                        QgsWkbTypes.MultiLineString),
                          ['MultiLineString ((1 1, 1 2, 2 2, 1 1),(3 3, 4 3, 4 4, 3 3))'])
 
+        # Missing z values
+        self.assertEqual(coerce_to_wkt('Point (1 1)', QgsWkbTypes.PointZ), ['PointZ (1 1 nan)'])
```

Comment:
For the record, PostGIS uses 0 as the missing value when forcing dimensions:
```
strk=# select ST_AsEWKT(ST_Force3DZ('POINT(1 1)'::geometry)) force_z, ST_AsEWKT(ST_Force3DM('POINT(1 1)'::geometry)) force_m;
   force_z    |    force_m    
--------------+---------------
 POINT(1 1 0) | POINTM(1 1 0)
(1 row)
```
PostGIS will consider geometries with NaN values as invalid:
```
strk=# select valid, reason, ST_AsEWKT(location)  FROM ST_IsValidDetail('POINT(0 NaN 0)'::geometry);
 valid |       reason       |   st_asewkt    
-------+--------------------+----------------
 f     | Invalid Coordinate | POINT(0 NaN 0)
(1 row)
```

PostGIS will NOT catch NaN in Z as being invalid (but I've no idea how many warms will be out their cans):
```
strk=# select valid, reason, ST_AsEWKT(location)  FROM ST_IsValidDetail('POINT(0 0 NaN)'::geometry);
 valid | reason | st_asewkt 
-------+--------+-----------
 t     |        | 
(1 row)
```

All of the above happens with POSTGIS="3.3.0dev 3.2.0-347-gf26b2ce71"

Reply:
And Postgis doesn't allow NaN in WKT (at reading):

```sql
# SELECT St_AsText(ST_MakePoint(1, 2, 'NaN'));
     st_astext
-------------------
 POINT Z (1 2 NaN)
```

But:

```sql
# SELECT ST_GeomFromText('POINTZ(1 2 NaN)');
ERREUR:  parse error - invalid geometry
ASTUCE : "POINTZ(1 2 NaN)" <-- parse error at position 15 within geometry

# SELECT St_AsText(ST_GeomFromText(ST_AsText(ST_MakePoint(1, 2, 'NaN'))));
ERREUR:  parse error - invalid geometry
ASTUCE : "POINT Z (1 2 NaN)" <-- parse error at position 17 within geometry
```